crates/orchestrator/src/lib.rs

```diff
@@ -179,6 +203,50 @@ where
         Ok(self.config.clone())
     }
 
+    #[allow(clippy::cast_possible_truncation)]
+    fn post_my_public_key(
+        &mut self,
```

Comment:
Should we throw an error if the public key is already assigned?

Reply:
Added. Also added a hashset to record who has posted.